### PR TITLE
Remove /examples from the Numba repo.

### LINKED ISSUE #5605: Task: move examples to numba/numba-examples
- Move examples into github.com/numba/numba-examples
    - like np.einsum (https://github.com/numba/numba/pull/5514)
    - lower barrier to submit examples
    - easier for user community to contribute
    - easier to test using our integration test

### PR DESCRIPTION
**Pull request by stuartarchibald**
This removes:
* `/examples` (is going to live in numba-examples)
*  `/benchmarks` (not sure it's needed)
* `/tutorials` (dated)

Then moves code from `/examples` that's set to be inlined into the docs in to unittests under `numba/tests/doc_examples` (refs in docs are fixed). These are true unittests and are run as part of the test suite, it makes sure the more involved doc examples actually work with the code base.

Fixes #5605

### CODE REVIEW

**Comment by stuartarchibald** [comment]
@esc could you take a look at this please? Thanks.

File: docs/source/developer/literal.rst [hunk 1]
```diff
@@ -45,9 +45,9 @@ use the following function:
 Code Example
 ~~~~~~~~~~~~
 
-.. literalinclude:: ../../../examples/literally_usage.py
+.. literalinclude:: ../../../numba/tests/doc_examples/test_literally_usage.py
    :language: python
-   :caption: from ``ex_literally_usage`` of ``examples/literally_usage.py``
+   :caption: from ``test_literally_usage`` of ``numba/tests/doc_examples/test_literally_usage.pyy``
```
[the remark this answers is not in the document]
Comment (esc) [reply docs/source/developer/literal.rst:50]: Yeah, this ends up a simply a typo in the caption:
Screen Shot 2020-05-15 at 17 00 34

[the remark this answers is not in the document]
Comment (stuartarchibald) [reply docs/source/developer/literal.rst:50]: Well caught.

**Comment by stuartarchibald** [comment]
> To review this I did the following:
> 
> a) check that the docs compile w/o error. This indicates that all files have been moved correctly, as a missing file would have caused an error with `sphinx` (I checked that too).
> b) read the changes carefully and found only one minor typo
> c) ran the tests and can confirm they all pass. I tried deliberately breaking them and that worked too.
> 
> The only other thing I noticed was that the following file has the execute bit set, not sure if that is needed as the other ones don't have it:
> 
> numba/tests/doc_examples/test_examples.py

Thanks for doing that. Typo is fixed. Expect the execute bit came from the original file I copied in from `/examples`, I've made it `0o0664` now.

**Comment by stuartarchibald** [comment]
@sklam are you happy with these changes?

**Comment by stuartarchibald** [comment]
Thanks for the review @esc

**Comment by sklam** [comment]
Thanks for doing this much needed repo cleanup. 

Before we merge, we should at least have a open PR in numba-examples repo that we can point folks to.

**Comment by stuartarchibald** [comment]
> Thanks for doing this much needed repo cleanup.

No problem.

> Before we merge, we should at least have a open PR in numba-examples repo that we can point folks to.

Done: https://github.com/numba/numba-examples/pull/17

**Pull request merged**